Sources/NIOSSH/SshAgent/Protocol.swift

```diff
@@ -0,0 +1,168 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// SSH Agent protocol as described here:
+// https://datatracker.ietf.org/doc/html/draft-miller-ssh-agent-17
+// and using the wire representation described here:
+// https://datatracker.ietf.org/doc/html/rfc4251#section-5
+//
+
+import Foundation
+import NIOCore
+
+enum MessageNumber: UInt8, Sendable {
+    case requestIdentities = 11
+    case signRequest = 13
+    case addIdentity = 17
+    case removeIdentity = 18
+    case removeAllIdentities = 19
+    case addSmartcardKey = 20
+    case removeSmartcardKey = 21
+    case lock = 22
+    case unlock = 23
+    case addIdConstrained = 25
+    case addSmartcardKeyContrained = 26
+    case messageExtension = 27
+    case failure = 5
+    case success = 6
+    case identitiesAnswer = 12
+    case signResponse = 14
+    case extensionFailure = 28
+    case extensionResponse = 29
+}
+
+/// A request to an SSH Agent
+///
+/// This enum handles the encoding of requests to `ByteBuffer`s
+public enum SshAgentRequest: Sendable {
+    case requestIdentities
+    case signRequest(keyBlob: [UInt8], data: [UInt8], flags: UInt32)
+    case addIdentity(Identity)
+
+    var messageNumber: MessageNumber {
+        switch self {
+        case .requestIdentities:
+            .requestIdentities
+        case .signRequest:
+            .signRequest
+        case .addIdentity:
+            .addIdentity
+        }
+    }
+
+    /// Encodes a SSH Agent request into a `ByteBuffer`
+    package func encode(into buf: inout ByteBuffer) {
+        buf.writeInteger(messageNumber.rawValue)
+
+        switch self {
+        case .signRequest(keyBlob: let blob, let data, let flags):
+            buf.writeInteger(UInt32(blob.count))
+            buf.writeBytes(blob)
+            buf.writeInteger(UInt32(data.count))
+            buf.writeBytes(data)
```

Comment:
The fixed width integer encoding use-case has helpers for this which we might want to use.